### ISSUE TITLE
[A11y][Grid] Interactive elements should meet WCAG 2.2 target-size requirements

### ISSUE THREAD

**Issue by dtopalov** [telerik/kendo-themes#5822]
## Feature Request Summary

Expand/Collapse icons in the Grid with Masted/Detail and Grouping enabled (and all other components with similar functionalities where the size of the interactive element is smaller) should comply with WCAG 2.2. target-size requirement of at least 24*24 px

## Problem Description
The size of the clickable expand/collapse elements is smaller by the required 24*24 px, e.g.:

Grid Master/Detail: (16*16)

<img width="771" height="784" alt="Image" src="https://github.com/user-attachments/assets/f05f448e-721c-4b1a-94f9-e25184f33119" />

Groupable Grid

<img width="771" height="784" alt="Image" src="https://github.com/user-attachments/assets/e46b9e71-1734-4076-a6e0-4859d348a694" />

## Proposed Solution

If possible, the interactive elements should be compliant with the target size requirement. 

It would be best to address the same issue in "compact/dense/small" Grids too, as the aim of the small size Grid is to allow displaying more items on the same visible surface area, rather than making all elements smaller. It seems the cell size of the high-density Grid is still big enough to accommodate a 24*24 interactive element in expand/collapse scenarios:

<img width="771" height="813" alt="Image" src="https://github.com/user-attachments/assets/09c52ce5-6ecf-403a-940c-d4efc3aa74a6" />

## Theme Scope

Which theme(s) should this feature apply to?

- [x] All themes

## Component/Area

Which component or area does this feature request affect?

- [ ] Specific component (please specify): All componens with interactive elements for expand/collapse functionality that do not meet the target-size requirement.

## Use Case

We should be compliant with WCAG 2.2 requirements.

## Design Considerations

Are there any design specifications or mockups?

## Implementation Notes

Do you have suggestions for implementation?

## Breaking Changes

Would this feature introduce breaking changes?

- [ ] Yes - please explain: ___________
- [ ] No
- [ ] Unsure

## Alternatives Considered

Describe any alternative solutions or features you have considered

## Additional Context

Any other context, screenshots, or examples

## Acceptance Criteria

What would constitute a successful implementation?

- [ ] Interactive targets for pointer interaction should have size of 24*24 px
---
